web/src/toga_web/widgets/timeinput.py

```diff
@@ -0,0 +1,80 @@
+import datetime
+
+from toga_web.libs import create_proxy
+
+from .base import Widget
+
+
+def py_time(native_time):
+    if isinstance(native_time, datetime.datetime):
+        return native_time.time().replace(microsecond=0)
+
+    if isinstance(native_time, str):
+        parsed_time = datetime.time.fromisoformat(native_time)
+        return parsed_time.replace(microsecond=0)
+
+    return datetime.datetime.now().time().replace(microsecond=0)
```

Comment:
Given there's a consistent "transformation" being applied here (in the form of the `.replace()` call), it's generally a good idea to try and restructure the logic so that it's only  implemented once. Since we know the type of `native_time` will be mutually exclusive, you can do this by using the `if` statements to convert to a consistent data type (`datetime.time`), and then do the `replace()` step on the return.